server/schedule/operator/create_operator.go

```diff
@@ -87,6 +87,15 @@ func CreateMovePeerOperator(desc string, cluster opt.Cluster, region *core.Regio
 		Build(kind)
 }
 
+// CreateReplaceLeaderPeerOperator creates an operator that replaces an old peer with a new peer, and move leader from old store firstly.
+func CreateReplaceLeaderPeerOperator(desc string, cluster opt.Cluster, region *core.RegionInfo, kind OpKind, oldStore uint64, peer *metapb.Peer, leader *metapb.Peer) (*Operator, error) {
+	return NewBuilder(desc, cluster, region).
+		RemovePeer(oldStore).
+		AddPeer(peer).
+		SetLeader(leader.GetStoreId()).
```

Comment:
move leader from old store firstly？

Reply:
Yes, you can check the test. verify the step 0 of the operator is transferred from theold store.